extension/src/providers/gotoProvider.ts

```diff
@@ -4,60 +4,40 @@ import { AutoLispExt } from '../extension';
 import { Sexpression } from '../format/sexpression';
 import { ReadonlyDocument } from '../project/readOnlyDocument';
 import { escapeRegExp } from "../utils";
-
+import { SearchPatterns, SearchHandlers } from './providerShared';
 
 export class AutolispDefinitionProvider implements vscode.DefinitionProvider{
 	async provideDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.Location | vscode.Location[]> {
 		const editor: vscode.TextEditor = vscode.window.activeTextEditor;
 		const rDoc = ReadonlyDocument.getMemoryDocument(document);
-		let selected: string =  editor.document.getText(editor.selection);
+		let selected: string = editor.document.getText(editor.selection);
 		if (selected === '') {
```

Comment:
I think it needs to remove this checking, it should always to check if it is in comment atom

Reply:
I think there is one bug:
```
(defun testcomment () 
  (progn 
    (progn 
      (setq dwgname (nth ixx (nth 2 GBL_wd_prj_ixlst)))
    )
  )
  (progn 
    ;                        (setq dwgname "" iec_p "") ;;F12 on dwgname 
  )
)
```

Reply:
I agree it shouldn't be working with comments, but we are talking about situation that someone visually saw it was a comment, chose to completely highlight it and to hit F12. Does that not suggest it did what they were hoping it would do for them?

Reply:
I tried C++ codes in vscode and MS C++ extension. It ignores the comment case.